.azure/scripts/process_metadata.py

```diff
@@ -46,12 +48,42 @@ def collect_metadata_and_thumbnails():
                     continue
 
                 # check if a quick deployable sample has a valid image URL
-                image_url = data.get('imageUrl')
-                if image_url:
-                    if not metadata_validator.validate_image_url(image_url):
-                        print(f"Warning: 'imageUrl' is not a valid image URL for the sample: {meta_file}. Excluding from index.json.")
+                image_version = data.get('imageVersion')
+                if image_version:
+                    if not metadata_validator.validate_image_version(image_version):
+                        print(f"Warning: 'imageVersion' is not a valid image version for the sample: {meta_file}. Excluding from index.json.")
                         continue
 
+                    if not metadata_validator.is_component_type_quick_deployable(component_type):
+                        print(f"Warning: '{component_type}' is not a quick deployable component type for directory: {meta_file}. Excluding from index.json.")
+                        continue
+
+                    display_name = data.get('displayName')
+                    image_name = display_name.strip().lower().replace(' ', '-')
+                    image_url = f"{CHOREO_ACR_BASE_URL}/samples/{image_name}:{image_version}"
+                    data['imageUrl'] = image_url
+
+                    # Attempt to pull the image from ACR
+                    pull_command = ['docker', 'pull', image_url]
+                    pull_result = subprocess.run(pull_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Since we are failing the run if this fails, do we need to use `subprocess` here?